presto-main/src/main/java/com/facebook/presto/sql/ExpressionUtils.java

```diff
@@ -104,7 +103,17 @@ public static Expression binaryExpression(LogicalBinaryExpression.Type type, Col
     {
         requireNonNull(type, "type is null");
         requireNonNull(expressions, "expressions is null");
-        Preconditions.checkArgument(!expressions.isEmpty(), "expressions is empty");
```

Comment:
Maybe provide another method instead of changing this one. I think there are cases where it would be better to fail when `expressions.isEmpty()`

Reply:
What are the cases? After this change, this method behaves correctly.

Reply:
I check usages and they all look safe. I added one more commit: `Remove combineConjunctsWithDefault helper method`